### DISCUSSION TITLE
Supabase Swift Initialization Error

### DISCUSSION THREAD

**Discussion by j3fff** [supabase/supabase#13740] [Questions] [answered]
Hello,

I am just trying to initialize Supabase in my Swift app, however am running into the following error at first declaration of the Supabase object.  I am taking right from the documentation, `let client = SupabaseClient(supabaseURL: "https://xyzcompany.supabase.co'", supabaseKey: "public-anon-key")`

The error I'm getting is `Cannot convert value of type 'String' to expected argument type 'URL'`.  I feel as though there must be something simple that I'm missing.  Has anyone else had this issue that they have been able to fix?  Thanks!

**Comment by GaryAustin1** [comment]
Looks like the documentation has a dangling single quote...
`"https://xyzcompany.supabase.co'"`
Try with out the `'` after `.co`.  
I assume you are putting your real supabase instance url in and not the example code.

**Reply by j3fff** [reply]
Sorry, I should have mentioned, I did try without the single quote and can confirm that I am using the URL from my project API settings.

**Reply by GaryAustin1** [reply, marked as answer]
I don't know Swift, but this is the code in the test example from the github...
```
 let client = SupabaseClient(
      supabaseURL: URL(string: "https://project-ref.supabase.co")!,
      supabaseKey: "ANON_KEY",
      options: SupabaseClientOptions(
        db: SupabaseClientOptions.DatabaseOptions(schema: customSchema),
        auth: SupabaseClientOptions.AuthOptions(storage: localStorage),
        global: SupabaseClientOptions.GlobalOptions(
          headers: customHeaders,
          httpClient: httpClient
        )
      )
    )
```
Seems to pass in the key for each value and then wrap the URL with addition typing.

This is quite different than the docs, so not sure what to tell you.
